### ISSUE TITLE
Relative path of images supported in .xsd file?

### ISSUE THREAD

**Issue by sreensr** [redhat-developer/vscode-xml#946]
![text] (absolute/path/to/image.png)" in xsd file works and image is visible when hover over the respective value. 
Unfortunately the ![text] (relative/path/to/image.png)" dosen't works. Is this a known issue?

**Comment by angelozerr** [comment]
Could you add a sample please.

**Comment by sreensr** [comment]
Sure. The following lines in .xsd file gives me the result as expected. Here I have used the absolute path of the imge. So when I hover over "Minion" I could see the image. 

<img width="754" alt="final_absolute" src="https://github.com/redhat-developer/vscode-xml/assets/150622128/69f15d44-a4bd-4a6a-9c01-0077c37fd6d6">

![results_with_absolute_path](https://github.com/redhat-developer/vscode-xml/assets/150622128/f779c5b0-2e43-4bf9-927b-dc1d38495637)

Unfortunately, when relative path of image is provided, result is not as expected and image is not visible when I hover over the corresponding value as shown in the picture below

<img width="626" alt="final_relative" src="https://github.com/redhat-developer/vscode-xml/assets/150622128/f40ef46f-2d68-4726-a6a2-03062bdf5c0c">


![results_with_relative_path](https://github.com/redhat-developer/vscode-xml/assets/150622128/d66f07d4-b330-4b8e-854f-b725a6666487)

My folder structure is:

<img width="107" alt="folder_structure" src="https://github.com/redhat-developer/vscode-xml/assets/150622128/b4370bb0-a769-47b2-9fbd-04540d6fe6d7">

.xsd file is inside the language_definition folder

**Comment by fbricon** [comment]
can you attach your test files here please?

**Comment by angelozerr** [comment]
@sreensr tobe honnest with you I didn't know that we support hover image like yoiu did.

Our hover doesn't update the content of the documentation before returning. We take the content as is and convert it to markdown.

To fix our issue we should pre process the uri of the markdown image.

**Comment by sreensr** [comment]
> can you attach your test files here please?

Sorry I won't be able to attach test files. It's confidential

**Comment by fbricon** [comment]
Surely you can come up with a simple, non-confidential test file.

**Comment by angelozerr** [comment]
@fbricon we don't do some process for documentation, in other words we return to  vscode the content `![text] (absolute/path/to/image.png)` which is a markdown syntax and it is vscode which renders the markdow.

To fix this issue we need to parse documentation content and update the relative path, it means that we need to implement a simple a Markdown parser which process Markdown link.

The only process that we do is to convert HTML to Markdown if the documentation uses HTML.

**Comment by fbricon** [comment]
Yes I was looking into it. The problem is passing the current path to the markdown processor will be a bit convoluted.

**Comment by angelozerr** [comment]
The first step that we could manage is to support relative path by writing html doc instead of markdown since remark have a convert with baseuri

**Comment by fbricon** [comment]
the fact markdown format seems supported in XSD documentation is a fluke. Given we need to replace the HTML->Markdown conversion library (we're also affected by https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2957), it most likely will stop working.

Writing HTML documentation (with `<img src=""`) is probably safer

**Comment by sreensr** [comment]
> The first step that we could manage is to support relative path by writing html doc instead of markdown since remark have a convert with baseuri

Because of some reason, the html version <img src="" simply doesn't works for images saved locally. And that's why I ended up using the markdown.

**Comment by angelozerr** [comment]
We need to investigate why it doesnt work but @fbricon is working to change the html > markdown converter library. Perhaps it will work better?

In short please dont use markdown.

**Comment by jtschea** [comment]
Dear vscode-xml-Developers,

I would be also interested in this feature. Just to emphasize that again, even if only using `<img src=.../>` it is just working for `http`/`https` but not for `file` URLs.

When being used with `file` URL the result is as follows:
```json
Result: {
    "contents": {
        "kind": "markdown",
        "value": "# ![Image]()\n\n#\n\nSource: [my.xsd](file:/path/to/my.xsd)"
    },
    "range": {
        "start": {
            "line": 20,
            "character": 42
        },
        "end": {
            "line": 20,
            "character": 79
        }
    }
}
```
In contrast to using `http`/`https` the generated Markdown does not have a valid path `![Image]()`.

Do you know a way to work around this?

Best regards and thanks in advance,
Jupp

**Comment by angelozerr** [comment]
@jtschea there is no solution for the moment, because we needto support base uri (xsd file base uri) to manage your usecase. 

We need to find time to work on this issue.

**Comment by jtschea** [comment]
Short update on this issue. I was able to work-around the missing possibility to link local files by spawning a http server hosting the images. Exemplary schema file:
```xml
<xs:annotation>
  <xs:documentation><img src="http://localhost/path/to/image.png"/></xs:documentation>
</xs:annotation>
```
Unfortunately, the markdown formatted tooltip is still not showing the image. I am assuming that this is a security issue not allowing access to potentially insecure URLs.

Referencing images using the Markdown editor was initially also not showing the image in the preview. After setting VSCode to allow insecure access using `Change preview security setting` the image was showing up correctly in the preview. This setting has no effect on the tooltip rendering where the image is still not shown.

Do you know a VSCode setting that has an effect on tooltips?

Best regards and thanks in advance,
Jupp

**Comment by angelozerr** [comment]
@jtschea I don't know but we need fix correctly the problem. The first step is to merge this PR https://github.com/eclipse/lemminx/pull/1607 which uses an another markdown parser.

**Comment by jtschea** [comment]
@angelozerr, thank you very much for your answer. I totally agree with you fixing this using a sustainable solution. Unfortunately, we need a short-term solution to this problem. A work-around would be very acceptable for us as well until there's a correct solution.

Do you potentially have a hint how to enable http access to localhost from Markdown tooltips?

**Comment by angelozerr** [comment]
> Do you potentially have a hint how to enable http access to localhost from Markdown tooltips?

No sorry and we discuss with @fbricon about documentation and our plan is to support HTML syntax and not markdown syntax for the first step.
